File: code_removed_clones/queue/java/262.java
package edu.bridgeport.mohammad;

public class Queue<T> {
  private Node<T> front;
  private Node<T> rear;

  public boolean enqueue(T item){
    Node<T> node = new Node<T>(item);

    if(front == null) {
      front = rear = node;
    } else {
      rear.setNext(node);
      rear = node;
    }

    return true;
  }

  public T look() {
	  if(front != null) {
		  return front.getData();
	  } else {
		  return null;
	  }
  }

  public T dequeue() {
    if(front == null) return null;
    Node<T> current = front;
    this.front = current.getNext();
    return current.getData();
  }

  public boolean empty() {
	  return front == null;
  }

  public String toString() {
	if(empty()) { return "Queue: [empty]"; }
	  
    StringBuilder output = new StringBuilder();
    output.append("Queue: ");

    Node<T> current = front;
    while(current != null) {
      output.append(current.getData() + ", ");
      current = current.getNext();
    }
    
    output.delete(output.length() - 2, output.length());    
    return output.toString();
  }
}

File: code/queue/cpp/262.c
#include "Queue.h"
#include <list>
#include <assert.h>

Queue::Queue() {
  q = new std::list<int>;
}

Queue::~Queue(){
  delete q; 
}

void Queue::enqueue(int value) {
  q->list::push_back(value);
}

int Queue::dequeue() {
  assert(!q->empty());
  int temp;
  temp = q->list::front();
  q->list::pop_front();
  return temp;
}

int Queue::queueSize() {
  return q->list::size();
}

bool Queue::isEmpty() {
  return q->list::empty();
}
